### ISSUE TITLE
Realtime Collaboration not showing other users

### ISSUE THREAD

**Issue by haribole** [metamaps/metamaps#1125]
Hello

I have two users working on the same map, but the junto panel does not list the other user.

any thoughts on how to start troubleshooting this?

Thanks
Hari

**Comment by Connoropolous** [comment]
Whichever port the real-time server is serving on (should be 5000 I think),
you'll need to forward that out from the Vagrant box as well...

The socket.io server drives a lot of the real-time interactive stuff

If it's not connecting to that server an error will show in the JS console

**Comment by haribole** [comment]
I installed the local copy outside of vagrant, since the vagrant build was not stable and would keep crashing. Opened the port 5000 as well, and now I can send a message to the junto panel of the other user, but do not see the other user listed in my panel

**Comment by haribole** [comment]
I see the following errors in the console:

![image](https://user-images.githubusercontent.com/8305237/28348763-b004cbfa-6c0c-11e7-8c03-e7a0c911c0e2.png)

**Comment by Connoropolous** [comment]
what if you browse to `40.143.89.232:5000` in another window, and see if it has a simple response of 'welcome to socket.io' or something similar. If so that's good. Then you need to change a value in your `.env` file... 
you would need to change
`export REALTIME_SERVER='http://localhost:5000'`
to
`export REALTIME_SERVER='http://40.143.89.232:5000'`

**Comment by haribole** [comment]
Could it have something to do with the fact that I am running this without vagrant, on the host machine?

**Comment by devvmh** [comment]
Yes.  You should point metamaps to the machine that has the working
realtime server.

Have you gotten a .env file setup? If sonyoi can point metamaps to the
working realtikd server by editing a line in the .env

**Comment by haribole** [comment]
I tried using both the internal IP within the private network address behind the firewall as well as the external exposed IP address and port, but no joy!

**Comment by devvmh** [comment]
If its running on your local machine,  try using localhost:5000 or
127.0.0.1:5000

**Comment by haribole** [comment]
Hi Devin, like I mentioned earlier, I tried both settings in .env, and the realtime server is accessible using telnet both from inside the firewall and outside. The browser console has the following error:

Failed to load resource: net::ERR_CONNECTION_REFUSED
localhost:5000/socket.io/?EIO=3&transport=polling&t=1500905638819-1 Failed to load resource: net::ERR_CONNECTION_REFUSED

Does that not mean that the junto panel on the browser end is looking to connect to the realtime server in localhost? Don't we need to specify the external IP address to the junto panel on my browser? Not sure if I am making sense.

**Comment by haribole** [comment]
I can try using browser on the same server where both the rails server and the realtime server is running to test this. Will update once I do this test.

**Comment by Connoropolous** [comment]
ah... I just noticed a mismatch between which ports are forwarding from the vagrant machine, and which port the realtime server is running on
https://github.com/metamaps/metamaps/blob/update-docs/Vagrantfile#L40 says 5001
and the realtime server is set to 5000

in your case @haribole you could run the realtime server using 
NODE_REALTIME_PORT=5001 node realtime/realtime-server.js
but we will change the docs to match up

**Comment by devvmh** [comment]
@haribole did this fix work for you?
